cmd/node/main.go

```diff
@@ -547,11 +544,25 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		fmt.Sprintf("%s_%s", defaultShardString, core.PathShardPlaceholder),
 		core.PathIdentifierPlaceholder)
 
-	pathManager, err := pathmanager.NewPathManager(pathTemplateForPruningStorer, pathTemplateForStaticStorer)
+	var pathManager *pathmanager.PathManager
+	pathManager, err = pathmanager.NewPathManager(pathTemplateForPruningStorer, pathTemplateForStaticStorer)
 	if err != nil {
 		return err
 	}
 
+	var currentEpoch uint32
+	var errNotCritical error
+	currentEpoch, errNotCritical = storageFactory.FindLastEpochFromStorage(
+		workingDir,
+		nodesConfig.ChainID,
+		defaultDBPath,
+		defaultEpochString,
+	)
+	if errNotCritical != nil {
```

Comment:
what happens if the blockchain is at epoch 3 and you start the node and no epoch can be found in the DB. Will it be able to request blocks older than 2 epochs and synchronize.

Reply:
this is not treated in this PR, this will be handled by the task for starting a node in a new shard.

Reply:
ok